katello_rapidnode.py

```diff
@@ -53,26 +56,56 @@ def cmd_debug(cmd):
 
 
 def paramiko_exec_command(system, username, password, command):
-    """Executes the command using paramiko"""
+    """Executes command using paramiko"""
     ssh = paramiko.SSHClient()
     ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
     ssh.connect(system, username=username, password=password)
     dummy, stdout, stderr = ssh.exec_command(command)
-    ret1 = stdout.read()
-    ret2 = stderr.read()
+    miko_stdout = stdout.read()
+    miko_stderr = stderr.read()
+    exit_status = stdout.channel.recv_exit_status()
     ssh.close()
-    return ret1, ret2
+    return miko_stdout, miko_stderr, exit_status
+
+
+def remote_cmd(host, username, password, command, halt_on_fail):
+    """Performs the actual paramiko goo and returns appropriate exit code"""
+    # std_results[0] = stdout
+    # std_results[1] = stderr
+    # std_results[2] = exit code
+    std_results = []
+    for results in paramiko_exec_command(host, username, password, command):
+        std_results.append(results)
+    print(std_results[0])
+    print(std_results[1])
```

Comment:
It's not really logs, though - it's virtually the same thing user would see in console if they ran the commands themselves.   For example, if I ran <tt>subscription-manager</tt>, I would get a result that tells me system registered (or didn't register) correctly, echoed to console. All this is doing is returning said context to user, as they would see it if they manually executed the command(s).